slack/src/main/java/com/hazelcast/jet/contrib/slack/SlackSinks.java

```diff
@@ -0,0 +1,170 @@
+                ctx -> new SlackContext(() -> HttpClients.custom().setRetryHandler(httpRetryHandler())
+                        .build(), accessToken, slackChannelId))
+                .<String>receiveFn((ctx, item) -> ctx.receiveFn(item)).destroyFn(ctx -> ctx.destroy()).build();
+    }
+
+    private static HttpRequestRetryHandler httpRetryHandler() {
+        return (exception, executionCount, context) -> {
+            if (executionCount >= RETRY_COUNT) {
+                // Do not retry if over max retry count
+                return false;
+            }
+            if (exception instanceof UnknownHostException) {
+                // Unknown host
+                return false;
+            }
+            if (exception instanceof ConnectTimeoutException) {
+                // Connection refused
+                return true;
+            }
+            if (exception instanceof HttpHostConnectException) {
+                // connection exception
+                return true;
+            }
+            if (exception instanceof IOException) {
+                // Interrupted Io exceptions
+                return true;
+            }
+            HttpClientContext clientContext = HttpClientContext.adapt(context);
+            HttpRequest request = clientContext.getRequest();
+            boolean idempotent = !(request instanceof HttpEntityEnclosingRequest);
+            if (idempotent) {
+                // Retry if the request is considered idempotent
+                return true;
+            }
+            return false;
+        };
+    }
+
+    private static final class SlackContext {
+
+        private final SupplierEx<CloseableHttpClient> httpClientSupplierEx;
+        private final CloseableHttpClient closeableHttpClient;
+        private final String accessToken;
+        private final String channel;
+
+        private SlackContext(SupplierEx<CloseableHttpClient> bulkRequestSupplier, String accessToken, String channel) {
+            this.httpClientSupplierEx = bulkRequestSupplier;
+            this.closeableHttpClient = bulkRequestSupplier.get();
+            this.accessToken = accessToken;
+            this.channel = channel;
+        }
+
+        private String receiveFn(String message) {
+            HttpPost request =  new HttpPost(URL);
+            CloseableHttpResponse response = null;
+            String result = "";
+            try {
+                // add request headers
+                request.addHeader("Authorization", String.format("Bearer %s", accessToken));
+                List<NameValuePair> urlParameters = new ArrayList<>();
+                urlParameters.add(new BasicNameValuePair("channel",  channel));
+                urlParameters.add(new BasicNameValuePair("text", message));
+                request.setEntity(new UrlEncodedFormEntity(urlParameters));
+                response = (CloseableHttpResponse) closeableHttpClient.execute(request);
+                result = EntityUtils.toString(response.getEntity());
```

Comment:
We return the result, but the returned value is ignored. Is there something worthwhile to check?

Reply:
Added the simple to pojo class to parse the response. In error scenarios sample message will look like this `{
    "ok": false,
    "error": "not_authed"
}` If the Response.ok false means then throwing the RuntimeException.